Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2872,30 +2875,21 @@ extension BrowserViewController {
     }
 
     /**
-     This adds the customSearchButton to the findInPage container
+     This adds the customSearchButton to the inputAssistant
      for cases where the inputAccessoryView could not be found for example
-     on the iPad where it does not exist.
+     on the iPad where it does not exist. However this only works on iOS9
      **/
-    func addCustomSearchButtonToFindInPage() {
-        let view = SearchAddBar()
-
-        view.addSubview(self.customSearchEngineButton)
-        findInPageContainer.addSubview(view)
-
-        view.snp_makeConstraints { make in
-            make.edges.equalTo(findInPageContainer)
-            make.height.equalTo(UIConstants.ToolbarHeight)
-        }
-
-        self.customSearchEngineButton.snp_remakeConstraints { make in
-            make.width.equalTo(view.snp_height)
-            make.height.equalTo(view.snp_height)
-            make.left.equalTo(view.snp_left).offset(20)
-            make.top.equalTo(view.snp_top)
+    func addCustomSearchButtonToInputAssistant(webContentView: UIView) {
+        if #available(iOS 9.0, *) {
+            guard customSearchBarButton == nil else {
+                return //The searchButton is already on the keyboard
+            }
+            let inputAssistant = webContentView.inputAssistantItem
+            let item = UIBarButtonItem(customView: customSearchEngineButton)
+            item.title = "SearchAdd" //we can check the label later to remove the correct button
```

Comment:
yea it would be the same label for the accessibility label. I was trying to think of other things to compare but that was the safest I could come up with. But because we use a customView on the button the title is never shown.
Although adding things to the inputAssistant is supported we dont actually control it. So weird issues of buttons getting cut off or crashes happen in certain cases.